test/node.js

```diff
@@ -68,4 +69,33 @@ describe('LevelDatastore', () => {
     expect(cids[0].version).to.be.eql(0)
     expect(cids).to.have.length(4)
   })
+
+  // The `.end()` method MUST be called on LevelDB iterators or they remain open,
+  // leaking memory.
+  //
+  // This test exposes this problem by causing an error to be thrown on process
+  // exit when an iterator is open AND leveldb is not closed.
+  //
+  // Normally when leveldb is closed it'll automatically clean up open iterators
+  // but if you don't close the store this error will occur:
+  //
+  // > Assertion failed: (ended_), function ~Iterator, file ../binding.cc, line 546.
+  //
+  // This is thrown by a destructor function for iterator objects that asserts
+  // the iterator has ended before cleanup.
+  //
+  // https://github.com/Level/leveldown/blob/d3453fbde4d2a8aa04d9091101c25c999649069b/binding.cc#L545
+  it('should not leave iterators open and leak memory', (done) => {
+    const cp = childProcess.fork(`${__dirname}/fixtures/test-level-iterator-destroy`, { stdio: 'pipe' })
+
+    let out = ''
+    cp.stdout.on('data', d => { out += d })
+    cp.stderr.on('data', d => { out += d })
+
+    cp.on('exit', code => {
+      expect(code).to.equal(0)
+      expect(out).to.not.include('Assertion failed: (ended_)')
```

Comment:
Is there a way to assert on the presence of something rather than the absence of something?  Even something we add to the fixture to demonstrate that the iterator has been torn down correctly?

Otherwise this test is coupled to the internal implementation of leveldown which may change without us knowing and this test may not expose the problem it is designed to.

Reply:
Not really, the leveldb iterator is not exposed an any way and the js iterator always ends fine. Given that this doesn’t even work on Node.js 10 I think this is not a good test.

That said I’ll all out of ideas for how to test this right now!

Reply:
(Still internal but at least it'd assert presence of something) you could check [`it._ended`](https://github.com/Level/abstract-leveldown/blob/52d0179e272f2f5c0420f3e24415510aa63e042b/abstract-iterator.js#L64).